forte/processors/pretrained_encoder_processors.py

```diff
@@ -0,0 +1,242 @@
+# Copyright 2019 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+
+import texar.torch as tx
+
+from texar.torch import HParams
+
+from forte.common.resources import Resources
+from forte.data.ontology.top import Annotation
+from forte.data.data_pack import DataPack
+from forte.processors.base import PackProcessor
+from forte.utils.utils import get_class
+
+
+__all__ = [
+    "PretrainedEncoder",
+]
+
+
+class PretrainedEncoder(PackProcessor):
+    r"""A wrapper of Texar pre-trained encoders.
+
+    This processor will compute the embedding vectors of the corresponding
+    annotations (determined by the user) by using the pre-trained encoders.
```

Comment:
This processor will compute the embedding vectors for entries of type `Annotation` using pre-trained models. The user can specify the pre-trained model type and the annotation class name via configuration. For the full list of pretrain models supported, see `default_config` for more details. The processor will add embedding vector for all entries matching the specified entry type.  The resulting vector can be accessed by the `embedding` field of the annotations.